internal/document_test.go

```diff
@@ -74,3 +113,448 @@ func TestDocument_Debug(t *testing.T) {
 		assert.ErrorContains(t, err, "PCR0 not found")
 	})
 }
+
+func TestDocument_Verify(t *testing.T) {
+	nitroDoc, debugNitroDoc, selfSignedDoc := initDocuments(t)
+	happyPathTests := []struct {
+		name         string
+		doc          internal.Document
+		certProvider internal.CertProvider
+	}{
+		{
+			name: "happy path - nitro",
+			doc:  nitroDoc,
+			certProvider: internal.NewNitroCertProvider(
+				internal.NewEmbeddedRootCertZipReader(),
+			),
+		},
+		{
+			name: "happy path - debug nitro",
+			doc:  debugNitroDoc,
+			certProvider: internal.NewNitroCertProvider(
+				internal.NewEmbeddedRootCertZipReader(),
+			),
+		},
+		{
+			name:         "happy path - self signed",
+			doc:          selfSignedDoc,
+			certProvider: internal.NewSelfSignedCertProvider(),
+		},
+	}
+
+	for _, tt := range happyPathTests {
+		t.Run(tt.name, func(t *testing.T) {
+			// when
+			_, _, err := tt.doc.Verify(
+				tt.certProvider,
+				internal.WithAttestationTime(),
+			)
+
+			// then
+			require.NoError(t, err)
+		})
+	}
+}
+func TestDocument_CheckMandatoryFields(t *testing.T) {
+	nitroDoc, _, _ := initDocuments(t)
```

Comment:
Didn't think it was necessary to run these tests on every type of attestation since they are all treated the same by the `CheckMandatoryFields` function

Reply:
Ok, I updated this and `CheckOptionalFields` to loop through happy tests on all attestation types